docs/enhancements/infrastructure-inventory-services-api/resource-server.sql

```diff
@@ -0,0 +1,139 @@
+CREATE DATABASE resource_server;
+
+-- Table: data_source
+CREATE TABLE IF NOT EXISTS data_source (
+    id            SERIAL PRIMARY KEY,
+    name          VARCHAR(255) NOT NULL,
+    generation_id INTEGER      NOT NULL DEFAULT 0,                 -- incremented on each audit/sync
+    last_snapshot TIMESTAMPTZ           DEFAULT CURRENT_TIMESTAMP, -- last completed snapshot
+    created_at    TIMESTAMPTZ           DEFAULT CURRENT_TIMESTAMP  -- TBD; tracks when first imported
+);
+
+-- Table: resource_type
+CREATE TABLE IF NOT EXISTS resource_type (
```

Comment:
will you keep a copy of the AlarmDictionary? But maybe a call to alarm server will do the trick. Basically what happens when client requests to list all the resourceType after the server is ready?

Reply:
Yes, as discussed we'll keep a cached copy of the alarm dictionary/definitions and refresh it periodically and based on notifications from the alarm server.  See diagram.